### PR TITLE
Feat place picker allow multi-select

### LINKED ISSUE #487: FE: User can pickup multiple location (SearchPage -> PlacePicker)

### PR DESCRIPTION
**Pull request by FilipMessa**
close #487
- refactor the SearchContext state:
	- `travelFrom` and `travelTo` into an arrays of locations
	- moved from `apps/core/components` folder back into `apps/core/scenes/search/`, because we use the PlacePiceker on one place and its directly connected to search context.
	- new setter and adder method for the SearchContext

PLS squash and merge or ask me before merge

NEXT STEPS
- the user can delete selected place with a keyboard one by one
- suggest travelFrom location base on a user location

### CODE REVIEW

File: apps/core/scenes/search/PlacePicker/PlacePickerContent.js [hunk 1]
```diff
@@ -31,11 +32,16 @@ type Props = {|
...
   +modalType: 'DESTINATION' | 'ORIGIN' | 'HIDDEN',
-  +setTravelFrom: (?Location) => void,
-  +setTravelTo: (?Location) => void,
+  +clearLocation: (type: 'travelTo' | 'travelFrom') => void,
+  +setLocation: (type: LocationSearchType, location: Location) => void,
+  +addLocation: (type: LocationSearchType, location: Location) => void,
   +setModalType: ('HIDDEN') => void,
 |};
 
+type State = {
```
Comment (jvaclavik) [inline apps/core/scenes/search/PlacePicker/PlacePickerContent.js]: You could use `{||}`

File: apps/core/scenes/search/PlacePicker/PlacePickerRenderer.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// @flow
+
+import * as React from 'react';
+import { QueryRenderer, graphql } from '@kiwicom/margarita-relay';
+
+import type { PlacePickerRendererQueryResponse } from './__generated__/PlacePickerRendererQuery.graphql';
+import PlacePickerContent from './PlacePickerContent';
+
+// @TODO s set the QueryRenderer variables base on closet location
```
Comment (JoseDuSV) [inline apps/core/scenes/search/PlacePicker/PlacePickerRenderer.js]: closet location -> closest location ?

File: apps/core/scenes/search/PlacePicker/PlacePickerContent.js [hunk 3]
```diff
@@ -31,11 +32,16 @@ type Props = {|
...
+  text: string,
```
Comment (RobinCsl) [inline apps/core/scenes/search/PlacePicker/PlacePickerContent.js]: `text` is very generic. Could you use a more descriptive name?

File: apps/core/scenes/search/PlacePicker/PlacePickerRenderer.js (hunk 2 above)
Comment (FilipMessa) [reply apps/core/scenes/search/PlacePicker/PlacePickerRenderer.js]: 👍

File: apps/core/scenes/search/PlacePicker/PlacePickerContent.js (hunk 3 above)
Comment (FilipMessa) [reply apps/core/scenes/search/PlacePicker/PlacePickerContent.js]: sure thanks I will try to find better name

File: apps/core/scenes/search/PlacePicker/PlacePickerContent.js (hunk 1 above)
Comment (FilipMessa) [reply apps/core/scenes/search/PlacePicker/PlacePickerContent.js]: 👍

Comment (FilipMessa) [reply apps/core/scenes/search/PlacePicker/PlacePickerContent.js]: done

File: apps/core/scenes/search/PlacePicker/PlacePickerContent.js (hunk 3 above)
Comment (FilipMessa) [reply apps/core/scenes/search/PlacePicker/PlacePickerContent.js]: done `searchText`

File: apps/core/scenes/search/PlacePicker/helpers.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 // @flow
 
-import { type Location } from '../../scenes/search/SearchContext';
+import { type Location } from '../SearchContext';
 
 export type OptionType = {
```
Comment (jvaclavik) [inline apps/core/scenes/search/PlacePicker/helpers.js]: Could we use here also `{||}`?

Comment (FilipMessa) [reply apps/core/scenes/search/PlacePicker/helpers.js]: I can't do that because I realize that we use the flow interface in the OptionPicker component.

Comment (FilipMessa) [reply apps/core/scenes/search/PlacePicker/helpers.js]: https://flow.org/en/docs/types/interfaces/

File: apps/core/scenes/search/Placepickers.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ import { MODAL_TYPE } from './SearchConstants';
 import PickersWrapper from './PickersWrapper';
 
 type Props = {|
-  +travelFrom: ?Location,
+  +travelFrom: ?Array<Location>,
   +travelTo: ?Location,
```
Comment (jvaclavik) [inline apps/core/scenes/search/Placepickers.js]: Why is it only for `travelFrom` and not `travelTo`?

Comment (FilipMessa) [reply apps/core/scenes/search/Placepickers.js]: nice shot I'll check why flow didn't warn me thanks.

**Pull request merged**